Extensions/CLIExtensionTemplate.s4ext

```diff
@@ -42,3 +42,6 @@ screenshoturls http://www.example.com/Slicer/Extensions/CLIExtensionTemplate/Scr
 
 # 0 or 1: Define if the extension should be enabled after its installation.
 enabled 1
+
+# Space separated list of tags indicating build system features that are required to build this extension
```

Comment:
Using NA value for dependencies caused lot of trouble and confusion in the past. I would recommend to not have "tags" row at all when there are no tags. If you want to keep the "tags" row in the template then you can make it a comment.

Reply:
I re-use the same CMake [code](https://github.com/adamrankin/Slicer/blob/0a356fda235bcb80f1c7863cfb7b889464c11946/Extensions/CMake/SlicerFunctionExtractExtensionDescription.cmake#L94) to parse a space separated list into a CMake list, and it already handles NA. 

If the tag line is empty, that is also handled and results in an empty extension_TAGS variable, which is fine.

Reply:
There would have to be at least one space after the tags in order for the CMake regex to work. I think NA is to ensure the regex works while simultaneously saying "no entries". The string parsing could be improved to detect an empty set (aka a line with just "tags\n" would work).

Reply:
I think the rule should be simply removing buildGroups line from s4ext file if there are no groups. If the buildGroups is there then it should be followed by valid group name(s).

Reply:
Ok, shouldn't be too hard.